pkg/kubelet/sysctl/safe_sysctls.go

```diff
@@ -51,19 +60,39 @@ var safeSysctlsIncludeReservedPorts = []string{
 // - it is namespaced in the container or the pod
 // - it is isolated, i.e. has no influence on any other pod on the same node.
 func SafeSysctlAllowlist() []string {
-	if goruntime.GOOS == "linux" {
-		// make sure we're on a new enough kernel that the ip_local_reserved_ports sysctl is namespaced
-		kernelVersion, err := getKernelVersion()
-		if err != nil {
-			klog.ErrorS(err, "Failed to get kernel version, dropping net.ipv4.ip_local_reserved_ports from safe sysctl list")
-			return safeSysctls
+	return getSafeSysctlAllowlist(getKernelVersion)
+}
+
+func getSafeSysctlAllowlist(getVersion func() (*version.Version, error)) []string {
+	if goruntime.GOOS != "linux" {
+		return nil
+	}
+
+	kernelVersion, err := getVersion()
+	if err != nil {
+		klog.ErrorS(err, "failed to get kernel version, unable to determine which sysctls are available")
+		return getSafeSysctlsNoRequiredByKernelVersion()
+	}
+
+	var safeSysctlAllowlist []string
+	for _, sc := range safeSysctls {
+		if sc.kernel != "" && kernelVersion.LessThan(version.MustParseGeneric(sc.kernel)) {
+			klog.ErrorS(nil, "Kernel version is too old, dropping the sysctl from safe sysctl list", "dropped sysctl", sc.name, "kernelVersion", kernelVersion)
+		} else {
+			safeSysctlAllowlist = append(safeSysctlAllowlist, sc.name)
 		}
-		if kernelVersion.LessThan(version.MustParseGeneric(ipLocalReservedPortsMinNamespacedKernelVersion)) {
-			klog.ErrorS(nil, "Kernel version is too old, dropping net.ipv4.ip_local_reserved_ports from safe sysctl list", "kernelVersion", kernelVersion)
-			return safeSysctls
+	}
+	return safeSysctlAllowlist
+}
+
+func getSafeSysctlsNoRequiredByKernelVersion() []string {
+	var safeSysctlsNoRequiredByKernelVersion []string
+	for _, sc := range safeSysctls {
+		if sc.kernel == "" {
+			safeSysctlsNoRequiredByKernelVersion = append(safeSysctlsNoRequiredByKernelVersion, sc.name)
 		}
 	}
-	return safeSysctlsIncludeReservedPorts
+	return safeSysctlsNoRequiredByKernelVersion
 }
 
 func getKernelVersion() (*version.Version, error) {
```

Comment:
I think @thockin refers to this ipvs code here https://github.com/kubernetes/kubernetes/pull/118846#pullrequestreview-1597195401

https://github.com/kubernetes/kubernetes/blob/8dc49c4b984b897d423aab4971090e1879eb4f23/pkg/proxy/ipvs/proxier.go#L546-L592

Reply:
A little confused,  Aren't we like that now? refer to https://github.com/kubernetes/kubernetes/blob/370c85f5ab0b0bfc3b30f235ddb040c246b0e1ff/pkg/kubelet/sysctl/safe_sysctls.go#L70

Reply:
I think the idea is to remove that dependency from that package, we can easily duplicate the code here and not depend on the interface

Reply:
better wait for him to confirm

Reply:
ahh😄,  maybe which is `shouldn't` rather than `should`..?

Reply:
/cc @thockin Can you confirm @aojea 's comment(https://github.com/kubernetes/kubernetes/pull/118846#discussion_r1309954882)?

Reply:
Updated, I got the point, Unnecessary `interface` affects performance.